src/containers/Transaction/Transaction.js

```diff
@@ -0,0 +1,143 @@
+import React, { useState, useEffect } from 'react';
+import Switch from '@material-ui/core/Switch';
+import TextField from '@material-ui/core/TextField';
+import InputLabel from '@material-ui/core/InputLabel';
+import MenuItem from '@material-ui/core/MenuItem';
+import FormControl from '@material-ui/core/FormControl';
+import Select from '@material-ui/core/Select';
+import Button from '@material-ui/core/Button';
+import { wrapper, heading, form, buttonContainer } from './transaction.module.scss';
+import Text from '../../components/Text';
+import useStockExchangeData from '../../hooks/useStockExchangeData';
+
+const Transaction = () => {
+  const [checked, setType] = useState(true);
+  const [currency, setCurrency] = useState('');
+  const [price, setPrice] = useState('');
+  const [amount, setAmount] = useState('');
+  const [total, setTotal] = useState('');
+
+  const [amountProps, setAmountProps] = useState({ error: false, helperText: '' });
+  const [currenciesProps, setCurrenciesProps] = useState({ error: false });
+
+  const [currencies, setCurrencies] = useState([]);
+  const stockExchangeData = useStockExchangeData();
+
+  const handleTypeChange = () => {
+    setType((checkedPrev) => !checkedPrev);
+  };
+
+  const handleCurrencyChange = (event) => {
+    setCurrenciesProps({ error: false });
+    setCurrency(event.target.value);
+    const currPrice = stockExchangeData.find(
+      (dataItem) => dataItem.name === event.target.value
+    ).price;
+    setPrice(currPrice);
+    if (amount !== '') {
+      setTotal(amount * currPrice);
```

Comment:
All those input setters should be "clear", without any extra operations. We can recalculate values of "each-dependent" inputs during re-rendering.